include/oc_log.h

```diff
@@ -53,6 +53,88 @@ extern "C" {
 #define OC_LOG_LEVEL_DEBUG_MACRO (7)
 #define OC_LOG_LEVEL_TRACE_MACRO (8)
 
+#ifdef __ANDROID__
+#include "port/android/oc_log_android.h"
+#define TAG "OC-JNI"
+#ifndef PRINT
+#define PRINT(...) __android_log_print(ANDROID_LOG_INFO, TAG, __VA_ARGS__)
+#endif /* !PRINT */
+#endif /* __ANDROID__ */
+
+#ifndef PRINT
+#include <stdio.h>
+#define PRINT(...) printf(__VA_ARGS__)
+#endif /* !PRINT */
+
+#ifndef SNPRINTF
+#include <stdio.h>
+#define SNPRINTF(...) snprintf(__VA_ARGS__)
+#endif /* !SNPRINTF */
+
+#define IPADDR_BUFF_SIZE 64 // max size : scheme://[ipv6%scope]:port = 63 bytes
+#define SNPRINT_ENDPOINT_ADDR(str, size, endpoint, addr_memb)                  \
+  do {                                                                         \
+    const char *scheme = "coap";                                               \
+    if ((endpoint).flags & SECURED)                                            \
+      scheme = "coaps";                                                        \
+    if ((endpoint).flags & TCP)                                                \
+      scheme = "coap+tcp";                                                     \
+    if ((endpoint).flags & TCP && (endpoint).flags & SECURED)                  \
+      scheme = "coaps+tcp";                                                    \
+    memset(str, 0, size);                                                      \
```

Comment:
we should include `<string.h>` for memset

Reply:
fixed